components/ContainedInputViews/src/MDCContainedInputView.h

```diff
@@ -0,0 +1,319 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+/**
+ A set of Contained Input View states outlined in the Material guidelines. These states overlap with
+ and extend UIControlState.
+ */
+typedef NS_OPTIONS(NSInteger, MDCContainedInputViewState) {
```

Comment:
Question/Clarification: Why are we using "Contained" in the name? How does "ContainedInput" is different than "Input"?

Reply:
The notion of "containers" comes from the [material.io guidelines for text fields](https://material.io/design/components/text-fields.html#anatomy). The first item under "Anatomy" talks about containers. It shows one that is filled and one that is outlined. The idea is these views are input views that can adopt these different container styles.

Reply:
MDCInputTextField is the text field that conforms to protocol. It has "TextField" in the name. If that is not enough to indicate that it is a text field, I imagine the course of action would be to make it very clear in the docs.